### ISSUE TITLE
Stopcasting and channelled spells

### ISSUE THREAD

**Issue by tii86** [SunwellTracker/issues#3234]
Decription: 

The macro /stopcasting does not work with regards to channelled spells such as penance.

How it should work: 

The macro /stopcasting should stop any cast, channelled or not, when used. On Frosthold, this means that one must use movement to cancel channelled spells, which introduces more lag.

**Comment by sth666** [comment]
Hello.

Try to use it :) propably it's works fine 🙃.

1st.
_Preventcasting when CHANNELING SPELL is channeling._
**#showtooltip
/stopmacro [channeling:CHANNELING SPELL]
/cast SPELL**

2nd.
_Prevent casting when ANY SPELL is channeling._

**#showtooltip
/stopmacro [channeling]
/cast SPELL**

Good Luck Bro!

**Comment by tii86** [comment]
I don't think you understand what I'm saying. For example, In the case of penance, in some situations you might only want 1 tick of penance, then cancel the cast. What I'm trying to achieve is canceling the cast (like one might cancel a flash heal when juking with /stopcasting).

I did try the macros you posted, however none of them worked.

**Comment by sth666** [comment]
I have just one more Idea :
Can try use : 

**/stopcasting [mod:shift] 
/cast [nomod] Spell**

/stopmacro [nomod:channeling]

/stopmacro [mod:shift, channeling]

/stopcasting [nomod] 
/cast [mod:shift] SPELL**

/stopcasting [mod:shift]  or other button 
/cast [nomod] SPELL**



try something like that :)

**Comment by sth666** [comment]
check website for simple macro explain.

I hopefully I helped little bit 

**http://www.macroexplain.com**
